fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/PruneOlapScanPartition.java

```diff
@@ -66,15 +68,21 @@ public Rule build() {
                     .stream()
                     .map(column -> scanOutput.get(column.getName().toLowerCase()))
                     .collect(Collectors.toList());
+            List<Long> manuallySpecifiedPartitions = scan.getManuallySpecifiedPartitions();
 
+            Map<Long, PartitionItem> idToPartitions;
+            if (manuallySpecifiedPartitions.isEmpty()) {
+                idToPartitions = partitionInfo.getIdToItem(false);
+            } else {
+                Map<Long, PartitionItem> allPartitions = partitionInfo.getAllPartitions();
+                idToPartitions = allPartitions.keySet().stream()
+                        .filter(id -> manuallySpecifiedPartitions.contains(id))
```

Comment:
we should pass isTemporary to here, because if user specify temporary partition manually, we should not select same name partition. for example:
```sql
select * from t temporary partitions(t1);
```
if t has partition t1, we should not select it.
if t has temporary partition t1, we should select it.

Reply:
如果 t1 是 temporary partition，那么 select * from t partitions(t1)会 报错，找不到t1: partition 't1' doesn't exist
反之，如果t1是partition，那么  select * from t temporary partitions(t1) 也会 报错